### PR TITLE
runtime-transaction no gate traits

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- gate-traits get in the way of implementing things as we make code generic

#### Summary of Changes
- get rid of gate traits. we already control the construction of this type, no need to gate

Fixes #

### CODE REVIEW

File: runtime-transaction/src/runtime_transaction.rs [hunk 1]
```diff
@@ -37,19 +37,7 @@ pub struct RuntimeTransaction<T> {
...
-trait DynamicMetaAccess: StaticMetaAccess {}
-
-// Implement the gate traits for the message types that should
-// have access to the static and dynamic metadata.
-impl StaticMetaAccess for SanitizedVersionedTransaction {}
-impl StaticMetaAccess for SanitizedTransaction {}
-impl DynamicMetaAccess for SanitizedTransaction {}
-
-impl<T: StaticMetaAccess> StaticMeta for RuntimeTransaction<T> {
+impl<T> StaticMeta for RuntimeTransaction<T> {
```
Comment (apfitzge) [inline runtime-transaction/src/runtime_transaction.rs:44]: static meta is available regardless of type.
access is gated via constructors.

File: runtime-transaction/src/runtime_transaction.rs [hunk 2]
```diff
@@ -66,7 +54,7 @@ impl<T: StaticMetaAccess> StaticMeta for RuntimeTransaction<T> {
     }
 }
 
-impl<T: DynamicMetaAccess> DynamicMeta for RuntimeTransaction<T> {}
+impl<T: SVMMessage> DynamicMeta for RuntimeTransaction<T> {}
```
Comment (apfitzge) [inline runtime-transaction/src/runtime_transaction.rs:61]: dynamic meta available for any type of SVMMessage, this that's where we know to have resolved ATLs

**Review by tao-stones — DISMISSED** [review]
lgtm - the gates were suspicious of over-engineering

**Comment by apfitzge** [comment]
rebased on conflict with #2987

**Review by tao-stones — APPROVED** [review]

**Pull request merged**